### PR TITLE
More cards modified to use less memory.

### PR DESCRIPTION
**Pull request by bafolts**
This introduces the change to extend `Card` for less memory consumption per instance of a card. This is for base cards starting with letter A through P.

### CODE REVIEW

**Comment by kberg** [comment]
That is A LOT of cards, my friend, with non-trivial differences in the diff. Any tips on how to make the UI do what I need? Or any chance you can break that up?

**Comment by bafolts** [comment]
Most of the diffs for the cards are only moving properties in to the call to super. Going to have to update hundreds of cards. Would need the UI to be smart enough to spot that the diff only involved moving the data. In lieu of the UI being able to spot that the code only moved how many cards in a batch would be preferred?

**Comment by kberg** [comment]
Actually hang out I think I have an idea.

File: src/cards/Card.ts [hunk 1]
```diff
@@ -2,32 +2,53 @@
...
+    this.properties = staticInstance;
+  }
+  public get adjacencyBonus() {
+    return this.properties.adjacencyBonus;
   }
   public get cardType() {
     return this.properties.cardType;
   }
+  public get cost() {
+    return this.properties.cost === undefined ? 0 : this.properties.cost;
```
Comment (kberg) [inline src/cards/Card.ts:47]: See, this seems like a good place for `this.properties.cost || 0`. What do you think?

**Review by kberg — COMMENTED** [review]
56 files reviewed, 109 to go.

File: src/cards/Card.ts (hunk 1 above)
Comment (bafolts) [reply src/cards/Card.ts:47]: Works for me. I am not too caught up on ternary or using `||` here. Six of one half a dozen of the other. Not having to check for `undefined` always better in my book.

**Review by kberg — COMMENTED** [review]
110 down, 55 to go.

File: tests/ares/AresHandler.spec.ts [hunk 2]
```diff
@@ -57,8 +59,8 @@ describe('AresHandler', function() {
     // Even though there's already a game, with a board, that laid out hazards, this is going to use a clean set-up.
 
     const deck = game.dealer.deck;
-    deck[deck.length - 1].cost = 5;
-    deck[deck.length - 2].cost = 3;
+    deck.push(new EnergyTapping());
```
Comment (kberg) [inline tests/ares/AresHandler.spec.ts:62]: The comment up top talks about how these cards are chosen for their predetermined costs. That was forced in the earlier version of the test, and now it's happenstance, and not well documented.

Can you please add assertions that those cards have cost 5 and 3? If not, I'd really prefer to restore something hand-crafted.

**Review by kberg — CHANGES_REQUESTED** [review]
That last one!

**Review by kberg — APPROVED** [review]
I can do that last one in another PR. I'd rather unblock you.

**Comment by bafolts** [comment]
This went out with the deployment today. I usually look at the memory charts every few hours to see if the server is still up. It has been improving over the past few weeks. With this change I expect the slow of the chart to decrease. Yesterday it took 12 hours before the memory climbed enough to get over 60% which seems to be where the heroku dyno crashes and the dyno runs `npm run start` again and we are back and running. If this change stretches that out to a full 24 hours to line up with when I normally deploy we will be in great shape. Besides the log messages I don't know what else would be climbing in memory with each game after this change. From what I saw with the log messages the node javascript should be using string interning so those messages should mostly end up as pointers to objects which shouldn't be more than a few bytes large. Thanks for approving! I will keep the subsequent cards to come to less than 26 at a time.

**Pull request merged**